checker_test.go

```diff
@@ -970,6 +975,179 @@ got args:
 want args:
   regexp
 `,
+}, {
+	about:   "ErrorAs: exact match",
+	checker: qt.ErrorAs,
+	got:     error(errBadWolf),
+	args:    []interface{}{errTestPtr()},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  bad wolf
+    file:line
+as:
+  &&quicktest_test.errTest{msg:"bad wolf", formatted:true}
+`,
+}, {
+	about:   "ErrorAs: wrapped match",
+	checker: qt.ErrorAs,
+	got:     fmt.Errorf("wrapped: %w", error(errBadWolf)),
+	args:    []interface{}{errTestPtr()},
+	expectedNegateFailure: `
+error:
+  unexpected success
+got:
+  e"wrapped: bad wolf\n  file:line"
+as:
+  &&quicktest_test.errTest{msg:"bad wolf", formatted:true}
+`,
+}, {
+	about:   "ErrorAs: fails if nil error",
+	checker: qt.ErrorAs,
+	got:     nil,
+	args:    []interface{}{errTestPtr()},
+	expectedCheckFailure: `
+error:
+  got nil error but want non-nil
+got:
+  nil
+as:
+  &(*quicktest_test.errTest)(nil)
+`,
+}, {
+	about:   "ErrorAs: fails if mismatch",
+	checker: qt.ErrorAs,
+	got:     errors.New("other error"),
+	args:    []interface{}{errTestPtr()},
+	expectedCheckFailure: `
+error:
+  error is not the expected error
```

Comment:
`want error not found in got error chain` ?

Reply:
It would be interesting to have a `chain:` note for this kind of errors in which we print the chain of got, for debugging. Not required here, but possibly a good idea for a follow-up?

Reply:
Updated.

Reply:
I'm not sure on adding a `chain` note. I fear that some of the errors in our code base may be deeply nested enough to cause a lot of noise. But perhaps that's not as true as I think…

Certainly worth considering in a follow-up though.